### PR TITLE
Fix generation of parameter symbols with explicit default values

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/65638

### CODE REVIEW

**Comment by weltkante** [comment]
Should it emit `default` instead of `null` on value types? The intellisense comment on `ExplicitDefaultValue` says
> Returns null if the parameter type is a struct and the default value of the parameter is the default value of the struct type.

or is the intention to have this case handled in the caller? As far as I can tell the `type` passed into it is currently just dropped when branching into the method emitting `null`, so still would not be entirely compatible to the way ExplicitDefaultValue is used in the symbols API.

(my use case: IIncrementalGenerator generating forwarding stubs to methods declared on a different object)

**Comment by CyrusNajmabadi** [comment]
Null is the value, not the literal emitted.

**Comment by CyrusNajmabadi** [comment]
> my use case: IIncrementalGenerator generating forwarding stubs to methods declared on a different object

SyntaxGenerator cannot be used in source generator scenarios.

**Comment by weltkante** [comment]
> Null is the value, not the literal emitted.

I now see I was looking at the VB version, C# does indeed differentiate. Sorry.

File: src/Workspaces/Core/Portable/Editing/SyntaxGenerator.cs [hunk 1]
```diff
@@ -273,10 +273,13 @@ public SyntaxNode ParameterDeclaration(IParameterSymbol symbol, SyntaxNode? init
             return ParameterDeclaration(
                 symbol.Name,
                 TypeExpression(symbol.Type),
-                initializer,
+                initializer is not null ? initializer :
```
Comment (akhera99) [inline src/Workspaces/Core/Portable/Editing/SyntaxGenerator.cs:276]: 👍

**Review by akhera99 — APPROVED** [review]
Looks good!

**Pull request merged**